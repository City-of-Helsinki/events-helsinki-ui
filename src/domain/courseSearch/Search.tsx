import classNames from 'classnames';
import { Button, IconHome, IconLocation, IconSearch } from 'hds-react';
import { uniq } from 'lodash';
import React, { FormEvent } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory, useLocation } from 'react-router';

import Checkbox from '../../common/components/checkbox/Checkbox';
import DateSelector from '../../common/components/dateSelector/DateSelector';
import MultiSelectDropdown from '../../common/components/multiSelectDropdown/MultiSelectDropdown';
import SearchAutosuggest from '../../common/components/search/SearchAutosuggest';
import SearchLabel from '../../common/components/search/searchLabel/SearchLabel';
import { AutosuggestMenuOption } from '../../common/types';
import { LinkedEventsSource } from '../../generated/graphql';
import useDivisionOptions from '../../hooks/useDivisionOptions';
import useLocale from '../../hooks/useLocale';
import IconRead from '../../icons/IconRead';
import { ROUTES } from '../app/constants';
import Container from '../app/layout/Container';
import {
  DEFAULT_SEARCH_FILTERS,
  EVENT_SEARCH_FILTERS,
} from '../eventSearch/constants';
import FilterSummary from '../eventSearch/filterSummary/FilterSummary';
import { getCategoryOptions } from '../eventSearch/utils';
import PlaceSelector from '../place/placeSelector/PlaceSelector';
import styles from './search.module.scss';
import { getSearchFilters, getSearchQuery } from './utils';

interface Props {
  scrollToResultList?: () => void;
}

const Search: React.FC<Props> = () => {
  const { t } = useTranslation();
  const locale = useLocale();
  const { push } = useHistory();
  const { search } = useLocation();
  const searchParams = React.useMemo(() => new URLSearchParams(search), [
    search,
  ]);

  const [autosuggestInput, setAutosuggestInput] = React.useState('');
  const [categoryInput, setCategoryInput] = React.useState('');
  const [divisionInput, setDivisionInput] = React.useState('');
  const [selectedCategories, setSelectedCategories] = React.useState<string[]>(
    []
  );
  const [selectedPlaces, setSelectedPlaces] = React.useState<string[]>([]);
  const [selectedTexts, setSelectedTexts] = React.useState<string[]>([]);
  const [placeInput, setPlaceInput] = React.useState('');
  const [selectedDateTypes, setSelectedDateTypes] = React.useState<string[]>(
    []
  );
  const [selectedDivisions, setSelectedDivisions] = React.useState<string[]>(
    []
  );
  const [start, setStart] = React.useState<Date | null>(null);
  const [end, setEnd] = React.useState<Date | null>(null);
  const [isCustomDate, setIsCustomDate] = React.useState<boolean>(false);
  const divisionOptions = useDivisionOptions();

  const { alsoOngoingCourses } = getSearchFilters(searchParams);

  const searchFilters = {
    alsoOngoingCourses,
    categories: selectedCategories,
    dateTypes: selectedDateTypes,
    divisions: selectedDivisions,
    places: selectedPlaces,
    text: selectedTexts,
    start,
    end,
  };

  // Initialize fields when page is loaded
  React.useEffect(() => {
    const {
      categories,
      dateTypes,
      divisions,
      places,
      text,
      end: endTime,
      start: startTime,
    } = getSearchFilters(searchParams);

    setSelectedCategories(categories);
    setSelectedDivisions(divisions);
    setSelectedPlaces(places);
    setSelectedTexts(text);
    setSelectedDateTypes(dateTypes);

    if (endTime || startTime) {
      setIsCustomDate(true);
    } else {
      setSelectedDateTypes(dateTypes);
    }
  }, [searchParams]);

  const clearInputValues = () => {
    setCategoryInput('');
    setDivisionInput('');
    setPlaceInput('');
    setAutosuggestInput('');
  };

  const clearFilters = () => {
    const search = getSearchQuery(DEFAULT_SEARCH_FILTERS);

    push({ pathname: `/${locale}${ROUTES.COURSES}`, search });

    clearInputValues();
  };

  const handleSubmit = (event?: FormEvent) => {
    if (event) {
      event.preventDefault();
    }

    moveToSearchPage();

    setAutosuggestInput('');
    // TODO: scroll
    // scrollToResultList();
  };

  const categories = getCategoryOptions(t);

  const moveToSearchPage = () => {
    const filters = {
      ...searchFilters,
      text: uniq([...searchFilters.text, autosuggestInput]).filter(
        (text) => text
      ),
    };
    const search = getSearchQuery(filters);

    push({ pathname: `/${locale}${ROUTES.COURSES}`, search });
  };

  const handleChangeDateTypes = (value: string[]) => {
    setSelectedDateTypes(value);
  };

  const toggleIsCustomDate = () => {
    setIsCustomDate(!isCustomDate);
  };

  const handleAlsoOngoingCoursesEventChange = (
    e: React.ChangeEvent<HTMLInputElement>
  ) => {
    const search = getSearchQuery({
      ...searchFilters,
      alsoOngoingCourses: e.target.checked,
    });

    push({ pathname: `/${locale}${ROUTES.COURSES}`, search });
  };

  const handleMenuOptionClick = async (option: AutosuggestMenuOption) => {
    const value = option.text;

    const { text } = getSearchFilters(searchParams);

    if (value && !text.includes(value)) {
      text.push(value);
    }

    const search = getSearchQuery({
      ...searchFilters,
      text,
    });

    setSelectedTexts(text);
    setAutosuggestInput('');

    push({ pathname: `/${locale}${ROUTES.COURSES}`, search });
    // TODO: handle scrolling
    // scrollToResultList();
  };

  return (
    <div className={styles.searchContainer}>
      <Container>
        <h1>{t('courseSearch.title')}</h1>
        <form onSubmit={handleSubmit}>
          <div className={styles.searchWrapper}>
            <div className={classNames(styles.gridRow, styles.autoSuggestRow)}>
              <div>
                <SearchLabel color="black" htmlFor="search">
                  {t('courseSearch.search.labelSearchField')}
                </SearchLabel>
                <SearchAutosuggest
                  name="search"
                  onChangeSearchValue={setAutosuggestInput}
                  onOptionClick={handleMenuOptionClick}
                  placeholder={t('courseSearch.search.placeholder')}
                  searchValue={autosuggestInput}
                />
              </div>
              <div>
                <MultiSelectDropdown
                  checkboxName="categoryOptions"
                  icon={<IconRead />}
                  inputValue={categoryInput}
                  name="category"
                  onChange={setSelectedCategories}
                  options={categories}
                  setInputValue={setCategoryInput}
                  showSearch={false}
                  title={t('eventSearch.search.titleDropdownCategory')}
                  value={selectedCategories}
                />
              </div>
            </div>
            <div className={classNames(styles.gridRow)}>
              <div className={styles.dateSelectorWrapper}>
                <DateSelector
                  dateTypes={selectedDateTypes}
                  endDate={end}
                  isCustomDate={isCustomDate}
                  name="date"
                  onChangeDateTypes={handleChangeDateTypes}
                  onChangeEndDate={setEnd}
                  onChangeStartDate={setStart}
                  startDate={start}
                  toggleIsCustomDate={toggleIsCustomDate}
                />
              </div>
              <div>
                <MultiSelectDropdown
                  checkboxName="divisionOptions"
                  icon={<IconLocation />}
                  inputValue={divisionInput}
                  name="division"
                  onChange={setSelectedDivisions}
                  options={divisionOptions}
                  selectAllText={t('eventSearch.search.selectAllDivisions')}
                  setInputValue={setDivisionInput}
                  showSearch={true}
                  showSelectAll={true}
                  title={t('eventSearch.search.titleDropdownDivision')}
                  value={selectedDivisions}
                />
              </div>
              <div>
                <PlaceSelector
                  checkboxName="placesCheckboxes"
                  icon={<IconHome />}
                  inputValue={placeInput}
                  name="places"
                  onChange={setSelectedPlaces}
                  selectAllText={t('eventSearch.search.selectAllPlaces')}
                  setInputValue={setPlaceInput}
                  showSearch={true}
                  showSelectAll={true}
                  title={t('eventSearch.search.titleDropdownPlace')}
                  value={selectedPlaces}
                  source={LinkedEventsSource.Linkedcourses}
                />
              </div>
            </div>
            <div className={classNames(styles.gridRow)}>
              <div>
                <Checkbox
                  checked={alsoOngoingCourses}
                  id={EVENT_SEARCH_FILTERS.ALSO_ONGOING_COURSES}
                  label={t('courseSearch.search.checkboxAlsoOngoingCourses')}
                  onChange={handleAlsoOngoingCoursesEventChange}
                />
              </div>
              <div className={classNames(styles.searchButtonWrapper)}>
                <Button
                  fullWidth={true}
                  iconLeft={<IconSearch />}
                  variant="success"
                  type="submit"
                >
                  {t('eventSearch.search.buttonSearch')}
                </Button>
              </div>
            </div>
            <FilterSummary onClear={clearFilters} route={ROUTES.COURSES} />
          </div>
        </form>
      </Container>
    </div>
  );
};

export default Search;
