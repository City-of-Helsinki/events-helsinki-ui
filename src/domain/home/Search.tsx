import { IconFill, IconFood } from "hds-react";
import React, { FunctionComponent } from "react";
import { useTranslation } from "react-i18next";
import { useHistory } from "react-router";

import { ReactComponent as CultureIcon } from "../../assets/icons/svg/culture.svg";
import { ReactComponent as DanceIcon } from "../../assets/icons/svg/dance.svg";
import { ReactComponent as MovieIcon } from "../../assets/icons/svg/movie.svg";
import { ReactComponent as MuseumIcon } from "../../assets/icons/svg/museum.svg";
import { ReactComponent as MusicIcon } from "../../assets/icons/svg/music.svg";
import { ReactComponent as NatureIcon } from "../../assets/icons/svg/nature.svg";
import { ReactComponent as SearchIcon } from "../../assets/icons/svg/search.svg";
import { ReactComponent as SportIcon } from "../../assets/icons/svg/sport.svg";
import { ReactComponent as TheatreIcon } from "../../assets/icons/svg/theatre.svg";
import Button from "../../common/components/button/Button";
import CategoryFilters from "../../common/components/category/CategoryFilters";
import DateSelector from "../../common/components/dateSelector/DateSelector";
import SearchAutosuggest from "../../common/components/search/SearchAutosuggest";
import SupriseMeButton from "../../common/components/search/SupriseMeButton";
import { AutosuggestMenuItem, Category } from "../../common/types";
import { CATEGORIES } from "../../constants";
import getLocale from "../../util/getLocale";
import { getSearchQuery } from "../../util/searchUtils";
import styles from "./search.module.scss";

const Search: FunctionComponent = () => {
  const { t } = useTranslation();
  const locale = getLocale();
  const [categories, setCategories] = React.useState<Category[]>([]);
  const [dateTypes, setDateTypes] = React.useState<string[]>([]);
  const [startDate, setStartDate] = React.useState<Date | null>(null);
  const [endDate, setEndDate] = React.useState<Date | null>(null);
  const [isCustomDate, setIsCustomDate] = React.useState<boolean>(false);
  const [searchValue, setSearchValue] = React.useState("");
  const { push } = useHistory();

  const handleCategoryClick = (newCategory: Category) => {
    if (
      categories.findIndex(category => category.value === newCategory.value) ===
      -1
    ) {
      setCategories([...categories, newCategory]);
    }
  };

  const handleChangeDateTypes = (value: string[]) => {
    setDateTypes(value);
  };

  const handleRemoveCategory = (removedCategory: Category) => {
    setCategories(
      categories.filter(category => category.value !== removedCategory.value)
    );
  };

  const toggleIsCustomDate = () => {
    setIsCustomDate(!isCustomDate);
  };

  const moveToSearchPage = React.useCallback(() => {
    const search = getSearchQuery({
      categories: categories.map(category => category.value),
      dateTypes,
      districts: [],
      endDate,
      isCustomDate,
      keywords: [],
      places: [],
      publisher: null,
      search: searchValue,
      startDate
    });

    push({ pathname: `/${locale}/events`, search });
  }, [
    categories,
    dateTypes,
    endDate,
    isCustomDate,
    locale,
    push,
    searchValue,
    startDate
  ]);

  const handleClickSupriseMe = () => {
    // TODO: Add suprise me feature
    alert("TODO: suprise me");
  };

  const handleMenuItemClick = (item: AutosuggestMenuItem) => {
    let search = "";
    switch (item.type) {
      case "district":
        search = getSearchQuery({
          categories: categories.map(category => category.value),
          dateTypes,
          districts: [item.value],
          endDate,
          isCustomDate,
          keywords: [],
          places: [],
          publisher: null,
          search: "",
          startDate
        });
        break;
      case "keyword":
      case "yso":
        search = getSearchQuery({
          categories: categories.map(category => category.value),
          dateTypes,
          districts: [],
          endDate,
          isCustomDate,
          keywords: [item.value],
          places: [],
          publisher: null,
          search: "",
          startDate
        });
        break;
      case "place":
        search = getSearchQuery({
          categories: categories.map(category => category.value),
          dateTypes,
          districts: [],
          endDate,
          isCustomDate,
          keywords: [],
          places: [item.value],
          publisher: null,
          search: "",
          startDate
        });
    }
    push({ pathname: `/${locale}/events`, search });
    setSearchValue("");
  };

  return (
    <>
      <div className={styles.searchContainer}>
        <SupriseMeButton onClick={handleClickSupriseMe} />
        <div className={styles.titleWrapper}>
          <h3>{t("home.search.title")}</h3>
        </div>
        <div className={styles.autosuggestWrapper}>
          <label>{t("home.search.labelSearchField")}</label>
          <SearchAutosuggest
            categories={categories}
            onChangeSearchValue={setSearchValue}
            onMenuItemClick={handleMenuItemClick}
            onRemoveCategory={handleRemoveCategory}
            placeholder={t("home.search.placeholder")}
            searchValue={searchValue}
          />
        </div>
        <div className={styles.dateSelectorWrapper}>
          <label>{t("home.search.labelDateRange")}</label>
          <DateSelector
            dateTypes={dateTypes}
            endDate={endDate}
            isCustomDate={isCustomDate}
            onChangeDateTypes={handleChangeDateTypes}
            onChangeEndDate={setEndDate}
            onChangeStartDate={setStartDate}
            startDate={startDate}
            toggleIsCustomDate={toggleIsCustomDate}
          />
        </div>
        <div className={styles.buttonWrapper}>
          <Button
            color="primary"
            fullWidth={true}
            iconLeft={<SearchIcon />}
            onClick={moveToSearchPage}
            size="default"
          >
            {t("home.search.buttonSearch")}
          </Button>
        </div>
      </div>
      <div className={styles.categoriesWrapper}>
        <CategoryFilters
          categories={[
            {
              icon: <MovieIcon />,
              text: t("home.category.movie"),
              value: CATEGORIES.MOVIE
            },
            {
              icon: <MusicIcon />,
              text: t("home.category.music"),
              value: CATEGORIES.MUSIC
            },
            {
              icon: <SportIcon />,
              text: t("home.category.sport"),
              value: CATEGORIES.SPORT
            },
            {
              icon: <MuseumIcon />,
              text: t("home.category.museum"),
              value: CATEGORIES.MUSEUM
            },
            {
              icon: <DanceIcon />,
              text: t("home.category.dance"),
              value: CATEGORIES.DANCE
            },
            {
              icon: <CultureIcon />,
              text: t("home.category.culture"),
              value: CATEGORIES.CULTURE
            },
            {
              icon: <NatureIcon />,
              text: t("home.category.nature"),
              value: CATEGORIES.NATURE
            },
            {
              icon: <IconFill />,
              text: t("home.category.influence"),
              value: CATEGORIES.INFLUENCE
            },
            {
              icon: <TheatreIcon />,
              text: t("home.category.theatre"),
              value: CATEGORIES.THEATRE
            },
            {
              icon: <IconFood />,
              text: t("home.category.food"),
              value: CATEGORIES.FOOD
            }
          ]}
          onClickCategory={handleCategoryClick}
        />
      </div>
    </>
  );
};

export default Search;
