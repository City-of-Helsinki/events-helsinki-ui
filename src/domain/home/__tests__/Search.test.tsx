import { MockedProvider } from "@apollo/react-testing";
import * as React from "react";
import { MemoryRouter } from "react-router";
import renderer from "react-test-renderer";

import Search from "../Search";

test("Search matches snapshot", () => {
  const component = renderer.create(
    <MockedProvider mocks={[]}>
      <MemoryRouter>
        <Search />
      </MemoryRouter>
    </MockedProvider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

export {};
