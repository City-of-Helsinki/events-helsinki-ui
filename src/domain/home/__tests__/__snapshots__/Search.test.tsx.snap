// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search matches snapshot 1`] = `
Array [
  <div
    className="searchContainer"
  >
    <button
      className="supriseMeButton"
      onClick={[Function]}
      type="button"
    >
      <div
        className="text"
      >
        Yllätä minut
      </div>
      <svg>
        suprise-me.svg
      </svg>
    </button>
    <div
      className="titleWrapper"
    >
      <h3>
        Löydä itsellesi tekemistä!
      </h3>
    </div>
    <div
      className="autosuggestWrapper"
    >
      <label>
        Mitä etsit?
      </label>
      <div
        className="searchAutosuggest"
        onClick={[Function]}
      >
        <div
          className="iconWrapper"
        >
          <svg>
            search.svg
          </svg>
        </div>
        <div />
        <div
          className="inputWrapper"
        >
          <input
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Aloita kirjoittamalla tähän, esim. teatteri"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="dateSelectorWrapper"
    >
      <label>
        Mille ajankohdalle etsit?
      </label>
      <div
        className="dateSelector"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          <div
            className="iconWrapper"
          >
            <svg>
              calendar.svg
            </svg>
          </div>
          <div
            className="info"
          >
            Valitse ajankohta
          </div>
          <div
            className="arrowWrapper"
          >
            <svg>
              angle-down.svg
            </svg>
          </div>
        </button>
      </div>
    </div>
    <div
      className="buttonWrapper"
    >
      <button
        className="_3wgeeaHxiHujNcKzeML5j4 _3aSBFzRPhZcNdul13_gERo _3bo_TEhSbadoDUbxUTgx4J _2AR2oE9bnVLLmiLEmYiWNc primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <div
          aria-hidden="true"
          className="_3Wlc6ydCOxrot7DuTOBW8x"
        >
          <svg>
            search.svg
          </svg>
        </div>
        <span
          className="_2zMLCVF6mmvV2UqfaBF5K"
        >
          Hae
        </span>
      </button>
    </div>
  </div>,
  <div
    className="categoriesWrapper"
  >
    <div
      className="categoryFilters"
    >
      <div
        className="categoryFilter"
      >
        <button
          onClick={[Function]}
        >
          <svg>
            movie.svg
          </svg>
          <span>
            Elokuva
          </span>
          <svg
            className=""
            height="100%"
            viewBox="0 0 199.404 199.404"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <polygon
                points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
              />
            </g>
          </svg>
        </button>
      </div>
      <div
        className="categoryFilter"
      >
        <button
          onClick={[Function]}
        >
          <svg>
            music.svg
          </svg>
          <span>
            Musiikki
          </span>
          <svg
            className=""
            height="100%"
            viewBox="0 0 199.404 199.404"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <polygon
                points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
              />
            </g>
          </svg>
        </button>
      </div>
      <div
        className="categoryFilter"
      >
        <button
          onClick={[Function]}
        >
          <svg>
            sport.svg
          </svg>
          <span>
            Liikunta ja urheilu
          </span>
          <svg
            className=""
            height="100%"
            viewBox="0 0 199.404 199.404"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <polygon
                points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
              />
            </g>
          </svg>
        </button>
      </div>
      <div
        className="categoryFilter"
      >
        <button
          onClick={[Function]}
        >
          <svg>
            museum.svg
          </svg>
          <span>
            Museot
          </span>
          <svg
            className=""
            height="100%"
            viewBox="0 0 199.404 199.404"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <polygon
                points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
              />
            </g>
          </svg>
        </button>
      </div>
      <div
        className="categoryFilter"
      >
        <button
          onClick={[Function]}
        >
          <svg>
            dance.svg
          </svg>
          <span>
            Tanssi
          </span>
          <svg
            className=""
            height="100%"
            viewBox="0 0 199.404 199.404"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <polygon
                points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
              />
            </g>
          </svg>
        </button>
      </div>
      <div
        className="categoryFilter"
      >
        <button
          onClick={[Function]}
        >
          <svg>
            culture.svg
          </svg>
          <span>
            Kulttuuri ja taide
          </span>
          <svg
            className=""
            height="100%"
            viewBox="0 0 199.404 199.404"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <polygon
                points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
              />
            </g>
          </svg>
        </button>
      </div>
      <div
        className="categoryFilter"
      >
        <button
          onClick={[Function]}
        >
          <svg
            className=""
            height="100%"
            viewBox="0 0 24 24"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.250564,11.2321141 C17.250564,10.8655334 17.497019,10.5435063 17.896874,10.0494683 C18.3886561,9.44432807 19,8.69086368 19,7.56461489 C19,5.62455688 17.3537705,4.8519175 16.469465,4.43627135 C16.2686916,4.3420883 16.0414115,4.23606188 15.9624557,4.17346117 C15.7690139,4.02288108 15.5626007,3.68111505 15.3437802,3.32017403 C14.7854496,2.39582662 13.940622,1 12,1 C10.0283597,1 9.21737029,2.40767 8.68159845,3.33765711 C8.4751853,3.69521431 8.2811795,4.03190461 8.09788914,4.1785369 C8.0206252,4.24057364 7.77247825,4.35505962 7.57339671,4.44755076 C6.673864,4.86263294 5,5.63640026 5,7.56461489 C5,8.69086368 5.61134386,9.44432807 6.10256204,10.0494683 C6.50354496,10.5435063 6.74887206,10.8655334 6.74887206,11.2321141 C6.74887206,11.5090235 6.66089268,11.6455044 6.44150822,11.9539961 C6.15162746,12.3640026 5.75402836,12.9240251 5.75402836,13.9662423 C5.75402836,15.6564615 7.1741057,16.7088302 8.54737351,17.7279246 C9.93699646,18.7577345 11.2493555,19.7305833 11.2493555,21.2645827 L11.2493555,21.2668385 L11.2493555,22.7540284 L12.7506445,22.7540284 L12.7506445,21.2645827 C12.7506445,19.7305833 14.0630035,18.7577345 15.4526265,17.7279246 C16.8253303,16.7088302 18.2459716,15.6564615 18.2459716,13.9662423 C18.2459716,12.9240251 17.8489365,12.3640026 17.5579278,11.9539961 C17.3391073,11.6455044 17.250564,11.5090235 17.250564,11.2321141 M16.3341121,12.8213825 C16.5726716,13.1586368 16.7452465,13.402272 16.7452465,13.9662423 C16.7452465,14.9013052 15.7323558,15.6519497 14.5592975,16.5227199 C13.9479536,16.9755881 13.3050274,17.4521431 12.7506445,18.0042701 L12.7506445,16.0771834 L15.0347245,13.7919755 L13.9738962,12.7311473 L12.7506445,13.9549629 L12.7506445,11.0764583 L15.5304544,8.29552046 L14.4696262,7.23412826 L12.7506445,8.95423783 L12.7506445,5.26417983 L11.2493555,5.26417983 L11.2493555,8.95423783 L9.53037383,7.23412826 L8.4695456,8.29552046 L11.2493555,11.0764583 L11.2493555,13.9549629 L10.0255398,12.7311473 L8.96527554,13.7919755 L11.2493555,16.0771834 L11.2493555,18.0042701 C10.6944086,17.4521431 10.0520464,16.9755881 9.44126652,16.5227199 C8.26764422,15.6519497 7.25418949,14.9013052 7.25418949,13.9662423 C7.25418949,13.402272 7.42676442,13.1586368 7.66532388,12.8213825 C7.92587818,12.4548018 8.24903319,11.9979858 8.24903319,11.2321141 C8.24903319,10.3139704 7.72792459,9.67160812 7.26772478,9.10481792 C6.85602643,8.59668063 6.50016113,8.15903964 6.50016113,7.56461489 C6.50016113,7.01530777 6.62592652,6.53818885 8.20278762,5.80897519 C8.53158234,5.65670319 8.8158234,5.52586207 9.03520786,5.34990332 C9.43901063,5.02674831 9.70182082,4.56993232 9.98042217,4.0871737 C10.5494683,3.09966162 10.9696262,2.5007251 12,2.5007251 C13.0213503,2.5007251 13.4595553,3.10304544 14.0601837,4.0967612 C14.3523203,4.57839188 14.6286658,5.03520786 15.0381083,5.35497905 C15.2507251,5.52135031 15.5203029,5.64824364 15.8316146,5.79431196 C16.8078472,6.25338382 17.4992749,6.64816307 17.4992749,7.56461489 C17.4992749,8.15903964 17.1439736,8.59668063 16.7317112,9.10481792 C16.2726394,9.67160812 15.7504028,10.3139704 15.7504028,11.2321141 C15.7504028,11.9979858 16.0741218,12.4548018 16.3341121,12.8213825"
            />
          </svg>
          <span>
            Luonto ja ulkoilu
          </span>
          <svg
            className=""
            height="100%"
            viewBox="0 0 199.404 199.404"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <polygon
                points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
              />
            </g>
          </svg>
        </button>
      </div>
      <div
        className="categoryFilter"
      >
        <button
          onClick={[Function]}
        >
          <svg
            className=""
            height="100%"
            viewBox="0 0 24 24"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.8294615,20.5124154 C16.4065385,20.5124154 17.2068462,20.6979538 17.9808462,20.8765692 C18.6911538,21.0420308 19.362,21.1971077 20.7417692,21.1971077 L20.7417692,22.9050308 C19.1681538,22.9050308 18.3678462,22.7201846 17.5945385,22.5422615 C16.8835385,22.3761077 16.2113077,22.2210308 14.8294615,22.2210308 C13.4476154,22.2210308 12.7760769,22.3761077 12.0643846,22.5415692 C11.2917692,22.7201846 10.4914615,22.9050308 8.91507692,22.9050308 C7.34007692,22.9050308 6.53907692,22.7201846 5.76507692,22.5415692 C5.05338462,22.3761077 4.38115385,22.2210308 3,22.2210308 L3,20.5124154 C4.57638462,20.5124154 5.37669231,20.6979538 6.15,20.8765692 C6.86169231,21.0420308 7.53392308,21.1971077 8.91507692,21.1971077 C10.2962308,21.1971077 10.9677692,21.0420308 11.6787692,20.8765692 C12.4527692,20.6979538 13.2530769,20.5124154 14.8294615,20.5124154 Z M3.90830769,14.7241 L6.39023077,17.2053308 L3,18.1136385 L3.90830769,14.7241 Z M18.9021692,3.86234615 L20.1164769,5.07665385 L8.57293846,16.6208846 L7.35863077,15.4065769 L18.9021692,3.86234615 Z M16.0396154,1 L17.2546154,2.215 L5.71038462,13.7585385 L4.49607692,12.5442308 L16.0396154,1 Z"
            />
          </svg>
          <span>
            Osallistu ja vaikuta
          </span>
          <svg
            className=""
            height="100%"
            viewBox="0 0 199.404 199.404"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <polygon
                points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
              />
            </g>
          </svg>
        </button>
      </div>
      <div
        className="categoryFilter"
      >
        <button
          onClick={[Function]}
        >
          <svg>
            theatre.svg
          </svg>
          <span>
            Teatteri
          </span>
          <svg
            className=""
            height="100%"
            viewBox="0 0 199.404 199.404"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <polygon
                points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
              />
            </g>
          </svg>
        </button>
      </div>
      <div
        className="categoryFilter"
      >
        <button
          onClick={[Function]}
        >
          <svg
            className=""
            height="100%"
            viewBox="0 0 24 24"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.9714253,10.9514872 C14.1495802,10.7554905 14.4021203,10.6437617 14.6669574,10.6437617 L15.7581797,10.6437617 C16.2917391,10.6437617 16.8001765,10.4170603 17.1567126,10.0201253 L21,5.74180501 L20.4183844,5.16018943 L17.5935184,7.98505543 L17.1781003,7.56963729 L19.9198676,4.66167257 L19.3382897,4.08009472 L16.4303627,6.821862 L16.0149068,6.40648158 L18.8397728,3.58161558 L18.258195,3 L13.9798747,6.84324964 C13.5829397,7.19982348 13.3562006,7.70826089 13.3562006,8.24182028 L13.3562006,9.4018829 C13.3562006,9.66513571 13.2458298,9.91628017 13.0519453,10.0943219 L11.7379445,11.309155 L3.64251505,3.18784912 C3.39091794,2.93621428 2.96044924,3.14269743 3.00292275,3.49606501 C3.24086497,5.47538366 4.30553191,7.76171113 6.54153997,9.99771918 L8.18653998,11.6427569 L9.95854562,11.6427569 L10.5907069,12.3698613 L4.03439548,18.4314657 L5.65114268,20.0338036 L11.6041869,13.5355065 L17.3328306,20.1243333 L18.9900521,18.5844608 L12.7310553,12.305472 L13.9714253,10.9514872 Z"
            />
          </svg>
          <span>
            Ruoka
          </span>
          <svg
            className=""
            height="100%"
            viewBox="0 0 199.404 199.404"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <polygon
                points="63.993,199.404 163.695,99.702 63.993,0 35.709,28.285 107.127,99.702 35.709,171.119  "
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
  </div>,
]
`;
